clang/test/CodeGen/constantexpr-fneg.c

```diff
@@ -8,7 +8,8 @@
 // CHECK:      entry:
 // CHECK-NEXT:   %retval = alloca i32
 // CHECK-NEXT:   store i32 0, ptr %retval
-// CHECK-NEXT:   [[ZEXT:%.*]] = zext i1 true to i32
+// CHECK-NEXT:   [[CMP:%.*]] = icmp ne ptr @b, @a
+// CHECK-NEXT:   [[ZEXT:%.*]] = zext i1 [[CMP]] to i32
```

Comment:
This looks to be a missing constant fold somewhere.  The compare should be folded to true, not a constantexpr.

Reply:
Oh, ignore me.  I didn't realize this was specific testing unsimplified IR.

Reply:
I remember looking into this test a while ago (the `zext i1 true` is pretty weird, usually IRBuilder would fold that away) and the context is that one of the values here gets RAUWed and that would enable the folding. For constant expressions, replacing an operand will also re-fold the expression. For instructions, this doesn't happen. In any case, it will get folded later if optimizations are enabled.